torch/csrc/jit/runtime/graph_executor.cpp

```diff
@@ -387,7 +387,9 @@ struct DifferentiableGraphOp {
     }
 
     detachVariables(*stack);
-    InterpreterState(f).run(*stack);
+    ExecutionPlan plan =
+        f.getPlanFor(*stack, GraphExecutor::getDefaultNumBailOuts());
+    InterpreterState(plan.code).run(*stack);
```

Comment:
Do I still need to `detachVariables`? Looks like for backward we are not doing that.

Reply:
Probably also need to add `
  last_executed_optimized_graph = plan.graph;`

Reply:
@jjsjann123 I believe we should be detaching them in forward, otherwise we might compute gradients twice (once via autodiff) and twice (through normal Variable autograd mechanics)